domain/application/state/application.go

```diff
@@ -3014,6 +3016,282 @@ WHERE application_uuid = $applicationID.uuid;
 	return hash.SHA256, nil
 }
 
+// GetApplicationConstraints returns the application constraints for the
+// specified application ID.
+// Empty constraints are returned if no constraints exist for the given
+// application ID.
+func (st *State) GetApplicationConstraints(ctx context.Context, appID coreapplication.ID) (constraints.Value, error) {
+	db, err := st.DB()
+	if err != nil {
+		return constraints.Value{}, internalerrors.Capture(err)
+	}
+
+	ident := applicationID{ID: appID}
+
+	query := `
+SELECT &applicationConstraint.*
+FROM application_constraint
+JOIN "constraint" ON "constraint".uuid = application_constraint.constraint_uuid
+LEFT JOIN container_type ON container_type.id = "constraint".container_type_id
+LEFT JOIN constraint_tag ON constraint_tag.constraint_uuid = "constraint".uuid	
+LEFT JOIN constraint_space ON constraint_space.constraint_uuid = "constraint".uuid	
+LEFT JOIN constraint_zone ON constraint_zone.constraint_uuid = "constraint".uuid	
+WHERE application_uuid = $applicationID.uuid;
```

Comment:
This looks like a good contender for a view

Reply:
Agree with @jack-w-shaw here.

Reply:
Yeah, I'm already doing it.

Reply:
Done, indeed it's way better as a view.